test/form.test.ts

```diff
@@ -2732,3 +2732,34 @@ test("form with thousandSeparator and decimalSeparator same value invalid", asyn
     });
   }).toThrow();
 });
+
+test("setValueAndUpdateRaw", async () => {
+  const M = types.model("M", {
+    foo: types.string
+  });
+
+  const o = M.create({ foo: "" });
+
+  const form = new Form(M, {
+    foo: new Field(converters.decimal())
+  });
+
+  const state = form.state(o, {
+    converterOptions: {
+      thousandSeparator: ",",
+      decimalSeparator: "."
+    }
+  });
+
+  // Setting the raw directly would update the value without relying on other event handlers
+  const field = state.field("foo");
+
+  await field.setRaw("1,000.00");
+  expect(field.raw).toEqual("1,000.00");
+  expect(field.value).not.toEqual("1,000.00");
+
+  // Instead, we set the value and update the raw based on the value
+  await field.setValueAndUpdateRaw("1,000.00");
+  expect(field.raw).toEqual("1,000.00");
+  expect(field.value).toEqual("1,000.00");
```

Comment:
I'm baffled that this even happens. Why is there a , in the value? It should be removed.